Compiler/src/abstractinterpretation.jl

```diff
@@ -3668,7 +3676,10 @@ function global_assignment_rt_exct(interp::AbstractInterpreter, sv::AbsIntState,
     if saw_latestworld
         return Pair{Any,Any}(newty, ErrorException)
     end
-    (valid_worlds, ret) = scan_partitions((interp, _, partition)->global_assignment_binding_rt_exct(interp, partition, newty), interp, g, sv.world)
+    newty′ = Ref{Any}(newty)
```

Comment:
```suggestion
    newty′ = Core.Box(newty)
```

I think that's a bit more canonical (esp for JET and other analysis tools)?

Reply:
JET can analyze both cases.

Since the frontend inserts `Core.Box`, are you suggesting we should consistently use `Core.Box` for captured variables?

I slightly prefer `Ref` because `.contents` is a bit long, but I don't have a strong preference.